modulemd/modulemd-module-stream.c

```diff
@@ -479,8 +496,8 @@ modulemd_module_stream_upgrade (ModulemdModuleStream *self,
         {
         case 1:
           /* Upgrade to ModuleStreamV2 */
-          updated_stream = modulemd_module_stream_upgrade_to_v2 (
-            current_stream, &nested_error);
+          updated_stream =
+            modulemd_module_stream_upgrade_to_v2 (current_stream);
           if (!updated_stream)
             {
               g_propagate_error (error, g_steal_pointer (&nested_error));
```

Comment:
> Now if we can just figure out what valgrind is complaining about. (I suppose re-ordering the includes could cause something odd to happen.)

Valgrind is choking on the `dup()` to `fcntl()` change. I'm trying to figure out why, but if it takes much longer I may just disable that [check](https://clang.llvm.org/extra/clang-tidy/checks/android-cloexec-dup.html).

When run under valgrind, `fcntl(fd, F_DUPFD_CLOEXEC)` returns `-1` and `EINVAL`. Which according to `fcntl(2)` means "the kernel does not recognize this value". (Since valgrind overrides fcntl() so it can track resources, it seems like it may be a bug in valgrind itself.)

Reply:
The issue is that the clang-tidy conversion isn't actually safe. It needs to have a third argument, zero. I've fixed that and it seems to work now.